### ISSUE TITLE
IBGP neighbors could be established for MLAG peers in each VRF when we set ‘overlay_mlag_rfc5549: true’

### ISSUE THREAD

**Issue by forcecity** [aristanetworks/ansible-avd#2906]
I just found some strange things when i tried to use ‘underlay_rfc5549: true’ and ‘overlay_mlag_rfc5549: true’.

Firstly, when i tried to set underlay_rfc5549 to true only, the overlay ibgp peers for mlag peers in each vrf are trying to use ipv4 addresses to establish ibgp neighbors (it will take an address from ‘mlag_peer_l3_ipv4_pool’), but in AF ipv4, it assigned command “neighbor <MLAG_PEER_GROUP> next-hop address-family ipv6 originate” (because of underlay mlag ibgp peers in vrf default) to only accept the updates which next-hop is ipv6 address, then, imported/networked routes will be dropped between each mlag-peer (error msg on the devices in syslog: because a local nexthop was not configured for afi/safi ipv6/unicast). So I couldn't see the routes from mlag peer device only (peer establishment is OK).

Secondly, If I set the var ‘overlay_mlag_rfc5549: true’ to deal with the issue above. Yes, it will change the overlay p2p mlag peer established in different vrfs via IPv6 link-local addr. But when I deployed the configs into devices, I found there are no bgp peers established in each VRFs anymore. I guess that’s because ipv6 unicast-routing is not enabled in each vrfs (there is only one command for vrf default: ‘ipv6 unicast-routing’). When I configured those commands ‘ipv6 unicast-routing vrf xxx’ manually, bgp peers are back in each VRFs. I think this/ese command/s should be generated automatically to support this variable parameter? For this point, i just had a quick workaround that we could modify "arista/avd/roles/eos_designs/python_modules/network_services/vrfs.py":

```
class VrfsMixin(UtilsMixin):
    <! --- omit some outputs  --- !> 
    def vrfs(self) -> dict | None:
        <! --- omit some outputs  --- !> 
        for tenant in self._filtered_tenants:
            for vrf in tenant["vrfs"]:
                <! --- omit some outputs  --- !> 
                if self._overlay_mlag_rfc5549 and self._mlag_ibgp_peering_enabled(vrf, tenant):
                    new_vrf["ip_routing_ipv6_interfaces"] = True
                    new_vrf["ipv6_routing"] = True
                else:
                    new_vrf["ip_routing"] = True
                <! --- omit some outputs  --- !> 
        if vrfs:
            return vrfs
```
It could generate the ipv6 routing vrf xxx when it has the corresponding vrfs and mlag overlay peers.

**Comment by ClausHolbechArista** [comment]
Thank you for reporting these issues. Can I ask which version of AVD you are using for these tests?

**Comment by forcecity** [comment]
> Thank you for reporting these issues. Can I ask which version of AVD you are using for these tests?

Now i'm using 3.8.6

**Comment by ClausHolbechArista** [comment]
I have recreated the scenarios and indeed there is something wrong here. I think the deployed cases have some IPv6 VLANs in use as well, so they don't see this.

For `underlay_rfc5549: true` alone, we will have to configure
```
router bgp <asn>
   vrf <vrfname>
      address-family ipv4
         no neighbor <mlag-peer> next-hop address-family ipv6
```
to prevent the overlay peering from inheriting from the common peer-group. Alternatively only set it on the underlay peer instead of the peer-group.

For `underlay_rfc5549: true` & `overlay_mlag_rfc5549` combined you are also correct that we are missing `ipv6 unicast-routing vrf <vrfname>`

We will discuss internally and submit a fix for this. My current thought is to always configure RFC5549 for MLAG peerings in overlay if it is enabled for underlay. So removing the `overlay_mlag_rfc5549` option.

**Comment by forcecity** [comment]
Thank you for your information. For your last point, I think it is better to de-couple the underlay and overlay's behavior. More choices for the endusers. So i suggest to leave ```overlay_mlag_rfc5549``` as it is now.

**Comment by ClausHolbechArista** [comment]
Thank you, that is good feedback.

**Comment by ClausHolbechArista** [comment]
So we will fix two things:
- If `underlay_rfc5549 is True and overlay_mlag_rfc5549 is False` configure
  ```eos
  router_bgp <asn>
     vrf <vrf_name>
        address-family ipv4
           no neighbor <mlag-peer-ip> next-hop address-family ipv6
  ```
- If `underlay_rfc5549 is True and overlay_mlag_rfc5549 is True` configure
  ```eos
  ipv6 unicast-routing vrf <vrf_name>
  ```

**Comment by forcecity** [comment]
LGTM

**Comment by carlbuchmann** [comment]
@forcecity I'm working on the PR to implement this fix, and we should have this merged in devel shortly and included in 4.0.0-rc3.

**Comment by carlbuchmann** [comment]
@forcecity Since this fix also introduced a new key in `eos_cli_config_gen`; we cannot backport/cherry-pick to 3.8.x train. This fix is included in 4.0.0 and above. If you need help upgrading to 4.x.x please let us know!
